vscode-dotnet-runtime-library/src/test/unit/WebRequestWorker.test.ts

```diff
@@ -18,82 +18,99 @@ import {
+        Debugging.log('Get Test Context.');
         const [eventStream, context] = getTestContext();
+
+        Debugging.log('Instantiate Install Script Worker.');
         const installScriptWorker: IInstallScriptAcquisitionWorker = new MockInstallScriptWorker(context, eventStream, true, true);
+
+        Debugging.log('Request the install script path.');
         const scriptPath = await installScriptWorker.getDotnetInstallScriptPath();
+
+        Debugging.log('Asserting the path is as expected.');
         assert.equal(scriptPath, path.join(__dirname, '..'));
+
+        Debugging.log('Scan the event stream events.');
         assert.exists(eventStream.events.find(event => event instanceof DotnetInstallScriptAcquisitionError));
         assert.exists(eventStream.events.find(event => event instanceof DotnetFallbackInstallScriptUsed));
     });
 
-    test('Install Script File Manipulation Failure', async () => {
+    test('Install Script File Manipulation Failure', async () => {
         const [eventStream, context] = getTestContext();
         const installScriptWorker: IInstallScriptAcquisitionWorker = new MockInstallScriptWorker(context, eventStream, true);
-        return assert.isRejected(installScriptWorker.getDotnetInstallScriptPath(), Error, 'Failed to Acquire Dotnet Install Script').then(() => {
-            assert.exists(eventStream.events.find(event => event instanceof DotnetInstallScriptAcquisitionError));
-        });
+        await assert.isRejected(installScriptWorker.getDotnetInstallScriptPath(), Error, 'Failed to Acquire Dotnet Install Script')
+        assert.exists(eventStream.events.find(event => event instanceof DotnetInstallScriptAcquisitionError));
     });
 
-    test('Web Requests Memoized on Repeated Installs', async () => {
+    test('Web Requests Cached on Repeated calls', async () => {
         const [eventStream, context] = getTestContext();
-        const webWorker = new MockWebRequestWorker(context, eventStream, '', 'MockKey');
-
-        // Make a request to cache the data
-        await webWorker.getCachedData(0);
-        const requests = [];
-        for (let i = 0; i < 10; i++) {
-            requests.push(webWorker.getCachedData(0));
-        }
-        for (const request of requests) {
-            assert.equal(await request, 'Mock Web Request Result');
-        }
+        const webWorker = new MockTrackingWebRequestWorker(context, eventStream, staticWebsiteUrl);
+
+        const uncachedResult = await webWorker.getCachedData();
+        // The data should now be cached.
+        const cachedResult = await webWorker.getCachedData();
+
+        assert.exists(uncachedResult);
+        assert.deepEqual(uncachedResult, cachedResult);
+
         const requestCount = webWorker.getRequestCount();
-        assert.isBelow(requestCount, requests.length);
-    });
+        assert.isAtMost(requestCount, 1);
+    }).timeout(maxTimeoutTime);
 
-    test('Web Requests are retried', async () => {

```

Comment:
Why did we remove the requests are retried test again?

Reply:
Ah yes: https://github.com/dotnet/vscode-dotnet-runtime/pull/799/commits/ed49b7d52dc571b568decddf3944669ce219cd5c

it was a faulty test. But it would be ideal to make a new retry test that isnt faulty. A bit hard to do with these libraries, though.